presto-main/src/main/java/com/facebook/presto/sql/planner/plan/AnalyzeFinishNode.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.plan;
+
+import com.facebook.presto.metadata.AnalyzeTableHandle;
+import com.facebook.presto.metadata.TableHandle;
+import com.facebook.presto.sql.planner.Symbol;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class AnalyzeFinishNode
```

Comment:
I don't like that the implementation is so coupled with the syntax in terms of concepts and modeling. The way to think about this is: the query engine supports some syntax for analyzing a table. It's implemented as some form of an aggregation query that updates the statistics of a table. 

Thus, there should be almost no reference to "analyze" in the planner/execution. In particular, this node should probably be called something like "WriteTableStatistics" or similar.